src/app/shared/github/github.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { EMPTY, Observable, of } from 'rxjs';
+import { delay, expand, map, toArray } from 'rxjs/operators';
+
+import { environment } from '~/environments/environment';
+import { GithubTag } from '~/app/shared/github/github.interfaces';
+import { tags } from './mocks/tags';
+import { compare } from './mocks/compare';
+import { associatedPullReqeusts } from './mocks/associated-pull-request';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class GithubService {
+  useMocks: boolean = false;
+  constructor(private http: HttpClient) {}
+
+  // TODO: GET ALL TAGS
+  getTags(page = 1): Observable<any> {
+    if (this.useMocks) {
```

Comment:
It feels somewhat weird to have code that handles mocks through production code. Consider using an interface for GithubService, with multiple implementations - 1 for production (this file) and 1 which will deal only with mock data. (It could be just me not seeing the purpose of it)

Reply:
This is for debugging, as you can quickly call the github api way to much with live-reload. You could properly make a prettier solution for this, but will leave as is for now,